x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -181,8 +186,35 @@ func (m *MongoCrypt) CreateExplicitEncryptionContext(doc bsoncore.Document, opts
 
 	algoStr := C.CString(opts.Algorithm)
 	defer C.free(unsafe.Pointer(algoStr))
-	if ok := C.mongocrypt_ctx_setopt_algorithm(ctx.wrapped, algoStr, -1); !ok {
-		return nil, ctx.createErrorFromStatus()
+
+	if opts.Algorithm == "Indexed" {
+		if ok := C.mongocrypt_ctx_setopt_index_type(ctx.wrapped, IndexTypeIndexed); !ok {
+			return nil, ctx.createErrorFromStatus()
+		}
+	} else if opts.Algorithm == "Unindexed" {
+		if ok := C.mongocrypt_ctx_setopt_index_type(ctx.wrapped, IndexTypeUnindexed); !ok {
+			return nil, ctx.createErrorFromStatus()
+		}
+	} else {
+		if ok := C.mongocrypt_ctx_setopt_algorithm(ctx.wrapped, algoStr, -1); !ok {
+			return nil, ctx.createErrorFromStatus()
+		}
+	}
+
+	if opts.QueryType != nil {
+		switch *opts.QueryType {
+		case options.QueryTypeEquality:
+			if ok := C.mongocrypt_ctx_setopt_query_type(ctx.wrapped, 1); !ok {
+				return nil, ctx.createErrorFromStatus()
+			}
+			break
```

Comment:
Go `switch` cases do not fall through by default, so this `break` is unnecessary.